packages/swapper/src/swappers/thorchain/utils/bitcoin/utils/getThorTxData.ts

```diff
@@ -0,0 +1,96 @@
+import { btcChainId } from '@shapeshiftoss/caip'
+import { bitcoin } from '@shapeshiftoss/chain-adapters'
+import { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import { Asset, BIP44Params, UtxoAccountType } from '@shapeshiftoss/types'
+
+import { SwapError, SwapErrorTypes } from '../../../../../api'
+import { bn, bnOrZero, fromBaseUnit, toBaseUnit } from '../../../../utils/bignumber'
+import { InboundResponse, ThorchainSwapperDeps } from '../../../types'
+import { getPriceRatio } from '../../getPriceRatio/getPriceRatio'
+import { makeSwapMemo } from '../../makeSwapMemo/makeSwapMemo'
+import { thorService } from '../../thorService'
+
+export const getThorTxInfo = async ({
+  deps,
+  sellAsset,
+  buyAsset,
+  sellAmount,
+  slippageTolerance,
+  destinationAddress,
+  wallet,
+  bip44Params,
+  accountType
+}: {
+  deps: ThorchainSwapperDeps
+  sellAsset: Asset
+  buyAsset: Asset
+  sellAmount: string
+  slippageTolerance: string
+  destinationAddress: string
+  wallet: HDWallet
+  bip44Params: BIP44Params
+  accountType: UtxoAccountType
+}) => {
+  try {
+    const { data: inboundAddresses } = await thorService.get<InboundResponse[]>(
+      `${deps.midgardUrl}/thorchain/inbound_addresses`
+    )
+
+    const btcInboundAddresses = inboundAddresses.find((inbound) => inbound.chain === 'BTC')
+
+    const vault = btcInboundAddresses?.address
+
+    if (!vault)
+      throw new SwapError(`[getPriceRatio]: vault not found for BTC`, {
+        code: SwapErrorTypes.RESPONSE_ERROR,
+        details: { inboundAddresses }
+      })
+
+    const priceRatio = await getPriceRatio(deps, {
+      buyAssetId: buyAsset.assetId,
+      sellAssetId: sellAsset.assetId
+    })
+
+    const expectedBuyAmount = toBaseUnit(
+      fromBaseUnit(bnOrZero(sellAmount).dividedBy(priceRatio), sellAsset.precision),
+      buyAsset.precision
+    )
+
+    if (
+      !bnOrZero(expectedBuyAmount).gte(0) ||
+      !(bnOrZero(slippageTolerance).gte(0) && bnOrZero(slippageTolerance).lte(1))
+    )
+      throw new SwapError('[makeTradeTx]: bad expected buy amount or bad slippage tolerance', {
+        code: SwapErrorTypes.BUILD_TRADE_FAILED,
+        details: { expectedBuyAmount, slippageTolerance }
+      })
+
+    const limit = bnOrZero(expectedBuyAmount)
+      .times(bn(1).minus(slippageTolerance))
+      .decimalPlaces(0)
+      .toString()
+
+    const memo = makeSwapMemo({
+      buyAssetId: buyAsset.assetId,
+      destinationAddress,
+      limit
+    })
+
+    const adapter = deps.adapterManager.get(btcChainId)
+
+    const pubkey = await (adapter as unknown as bitcoin.ChainAdapter).getPublicKey(
+      wallet,
+      bip44Params,
```

Comment:
https://github.com/shapeshift/lib/blob/main/packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts#L267

this might help? if we're just overriding `accountNumber`?